From 0c69d1b2ff5692687487bdbe41b0ced62a3a53ce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BC=A0=E7=8E=89=E6=96=8C?=
 <46096697+uuiid@users.noreply.github.com>
Date: Mon, 5 Dec 2022 13:42:15 +0800
Subject: [PATCH] Fix-depends

fix-directory-output
remove-owner-read-perms
---
 CMake/3rd_party_libs.cmake | 3 ++-
 CMake/config.cmake         | 5 ++---
 CMake/utility.cmake        | 2 +-
 doc/CMakeLists.txt         | 6 ++----
 4 files changed, 7 insertions(+), 9 deletions(-)

diff --git a/CMake/3rd_party_libs.cmake b/CMake/3rd_party_libs.cmake
index e5965f8..c6ad856 100644
--- a/CMake/3rd_party_libs.cmake
+++ b/CMake/3rd_party_libs.cmake
@@ -53,7 +53,8 @@ if (BUILD_BENCHMARKS OR BUILD_EXAMPLES)
     find_package(Threads REQUIRED)
 endif()
 
-set(RAPID_JSON_DIR ${RTTR_3RD_PARTY_DIR}/rapidjson-1.1.0)
+find_package(RapidJSON CONFIG REQUIRED)
+set(RAPID_JSON_DIR ${RAPIDJSON_INCLUDE_DIRS})
 set(NONIUS_DIR ${RTTR_3RD_PARTY_DIR}/nonius-1.1.2)
 
 # Prepare "Catch" library for other executables
diff --git a/CMake/config.cmake b/CMake/config.cmake
index 1b692f5..e465709 100644
--- a/CMake/config.cmake
+++ b/CMake/config.cmake
@@ -47,7 +47,7 @@ set(CMAKE_EXECUTABLE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin")
 
 # here we specify the installation directory
 if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-  set(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/install" CACHE PATH  "RTTR install prefix" FORCE)
+  #set(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/install" CACHE PATH  "RTTR install prefix")
 endif()
 
 # in order to group in visual studio the targets into solution filters
@@ -171,6 +171,5 @@ if (BUILD_INSTALLER)
             COMPONENT Devel)
 
     install(FILES "${LICENSE_FILE}" "${README_FILE}"
-            DESTINATION ${RTTR_ADDITIONAL_FILES_INSTALL_DIR}
-            PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
+            DESTINATION ${RTTR_ADDITIONAL_FILES_INSTALL_DIR})
 endif()
diff --git a/CMake/utility.cmake b/CMake/utility.cmake
index cd1e835..8e7a0c6 100644
--- a/CMake/utility.cmake
+++ b/CMake/utility.cmake
@@ -165,7 +165,7 @@ function(loadFolder FOLDER _HEADER_FILES _SOURCE_FILES)
     getNameOfDir(CMAKE_CURRENT_SOURCE_DIR DIRNAME)
     if (${shouldInstall})
       if (NOT ${FULL_HEADER_PATH} MATCHES ".*_p.h$") # we don't want to install header files which are marked as private
-        install(FILES ${FULL_HEADER_PATH} DESTINATION "include/${DIRNAME}/${REL_PATH}" PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
+        install(FILES ${FULL_HEADER_PATH} DESTINATION "include/${DIRNAME}/${REL_PATH}")
       endif()
     endif()
   endforeach()
diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index de70600..a92d642 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -132,12 +132,10 @@ set_target_properties(doc PROPERTIES FOLDER "Documentation")
 
 install(DIRECTORY "${DOXYGEN_OUTPUT_DIRECTORY}/${DOXYGEN_CUSTOM_HTML_DIR}"
         DESTINATION "${DOXYGEN_DOC_INSTALL_DIR}"
-        PATTERN "*.*"
-        PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
+        PATTERN "*.*")
 
 install(FILES "${DOXYGEN_OUTPUT_DIRECTORY}/${DOXYGEN_INDEX_FILE}"
-        DESTINATION "${DOXYGEN_DOC_INSTALL_DIR}"
-        PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
+        DESTINATION "${DOXYGEN_DOC_INSTALL_DIR}")
 
 #########################################################################################
 
-- 
2.37.3.windows.1

